lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1040,409 @@ spec = do
+        addrs <- listAddresses ctx wDest
+
+        let amt = 1 :: Natural
+        let destination = (addrs !! 1) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+
+        tx <- request @(ApiTransaction t) ctx (postTxEp wSrc) Default payload
+        expectResponseCode HTTP.status202 tx
+        expectEventually' ctx balanceAvailable amt wDest
+        expectEventually' ctx balanceTotal amt wDest
+
+        -- Verify Tx list contains Incoming and Outgoing
+        r <- request @([ApiTransaction t]) ctx (listTxEp wSrc mempty)
+            Default Empty
+        expectResponseCode @IO HTTP.status200 r
+
+        verify r
+            [ expectListItemFieldEqual 0 direction Outgoing
+            , expectListItemFieldEqual 1 direction Incoming
+            ]
+
+    -- This scenario covers the following matrix of cases. Cases were generated
+    -- using one of pairwise test cases generation tools available online.
+    -- +---+----------+----------+------------+--------------+
+    --     |  start   |   end    |   order    |    result    |
+    -- +---+----------+----------+------------+--------------+
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    --  17 | t1       | t1       | empty      | 1st one      |
+    --  18 | t2       | t2       | descending | 2nd one      |
+    -- +---+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03x - Can limit/order results with start, end and order"
+        $ \ctx -> do
+        let a1 = sum $ replicate 10 1
+        let a2 = sum $ replicate 10 2
+        w <- fixtureWalletWith ctx $ mconcat
+                [ replicate 10 1
+                , replicate 10 2
```

Comment:
Actually… what is this doing?

This is relying on [`We only have 10 UTxO available in each faucet`](https://github.com/input-output-hk/cardano-wallet/blob/37be502ca7443f53316b75caedf7623c16ce6f95/lib/core-integration/src/Test/Integration/Framework/DSL.hs#L794-L810) to make sure the wallet gets set up using two different transactions?

Reply:
That was exactly the intent (following [this](https://github.com/input-output-hk/cardano-wallet/pull/608#discussion_r313385024) suggestion).